core/src/main/java/io/grpc/Metadata.java

```diff
@@ -170,6 +170,13 @@ private void storeAdd(String name, MetadataEntry value) {
   }
 
   /**
+   * Returns the total number of headers in this metadata, including duplicates.
+   */
+  public int size() {
```

Comment:
I don't see this method as being generally useful. Let's mark it `@Internal` like serialize and the constructor.